File: src/main/java/com/drunkensoftball/api/bat/controller/BatController.java
package com.drunkensoftball.api.bat.controller;

import com.drunkensoftball.api.bat.domain.AtBat;
import com.drunkensoftball.api.bat.domain.BatRequest;
import com.drunkensoftball.api.bat.service.BatService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class BatController {

    @Autowired
    BatService batService;

    private static final String URL_BAT = "/bat";

//    @RequestMapping(value = URL_BAT, method = RequestMethod.POST)
//    public String batUpload(@RequestHeader(name = "Authorization") final String token, @RequestBody final BatRequest batRequest) {
//        return batService.addPlays(token, batRequest.getGameUuid(), batRequest.getRosterUuid(), batRequest.getAtBatResult(), batRequest.getUniqueId()).getUuid();
//    }

    @RequestMapping(value = URL_BAT, method = RequestMethod.POST)
    public Integer batBulkUpload(@RequestHeader(name = "Authorization") final String token, @RequestBody(required = true) final List<BatRequest> batRequestList) {
        int resultsAdded = 0;
        for(final BatRequest batRequest : batRequestList) {
            batService.addPlays(token, batRequest.getGameUuid(), batRequest.getRosterUuid(), batRequest.getAtBatResult(), batRequest.getUniqueId());
            resultsAdded +=1;
        }
        return resultsAdded;
    }

}

File: src/main/kotlin/com/drunkensoftball/api/bat/controller/BatController.kt
package com.drunkensoftball.api.bat.controller


import com.drunkensoftball.api.auth.service.AuthenticationRequired
import com.drunkensoftball.api.bat.domain.BatRequest
import com.drunkensoftball.api.bat.service.BatService
import com.drunkensoftball.api.user.domain.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod.POST
import org.springframework.web.bind.annotation.RestController

@RestController
class BatController {

    @Autowired
    lateinit var batService: BatService

    @AuthenticationRequired
    @RequestMapping(value = [URL_BAT], method = [POST])
    fun batBulkUpload(@AuthenticationPrincipal user: User, @RequestBody(required = true) batRequestList: List<BatRequest>): Any? {
        var resultsAdded = 0
        var resultsProcessed = 0

        for (batRequest in batRequestList) {
            resultsAdded += batService.addPlays(user, batRequest.gameUuid, batRequest.rosterUuid, batRequest.atBatResult, batRequest.uniqueId, batRequest.rbis)
            resultsProcessed += 1
        }
        return mapOf("newCount" to resultsAdded, "processedCount" to resultsProcessed)
    }

    companion object {

        private const val URL_BAT = "/bat"
    }

}
